osu.Game.Rulesets.Taiko/Difficulty/Preprocessing/Colour/TaikoColourDifficultyPreprocessor.cs

```diff
@@ -36,14 +37,22 @@ public static void ProcessAndAssign(List<DifficultyHitObject> hitObjects)
                     AlternatingMonoPattern monoPattern = repeatingHitPattern.AlternatingMonoPatterns[i];
                     monoPattern.Parent = repeatingHitPattern;
                     monoPattern.Index = i;
-                    monoPattern.FirstHitObject.Colour.AlternatingMonoPattern = monoPattern;
+
+                    foreach (var hitObject in monoPattern.AllHitObjects)
+                    {
+                        hitObject.Colour.AlternatingMonoPattern = monoPattern;
+                    }
 
                     for (int j = 0; j < monoPattern.MonoStreaks.Count; ++j)
                     {
                         MonoStreak monoStreak = monoPattern.MonoStreaks[j];
                         monoStreak.Parent = monoPattern;
                         monoStreak.Index = j;
-                        monoStreak.FirstHitObject.Colour.MonoStreak = monoStreak;
+
+                        foreach (var hitObject in monoStreak.HitObjects)
+                        {
+                            hitObject.Colour.MonoStreak = monoStreak;
+                        }
```

Comment:
Isn't it functionally equivalent to add:
```
hitObject.Colour.AlternatingMonoPattern = monoPattern;
hitObject.Colour.RepeatingPattern = repeatingHitPattern;
```
In this loop? Just to reduce the amount of looping.

Reply:
Yeah it should be, will apply this when I'm back

Reply:
Applied in 122064d03fcdb4fa65dd7ccbbb20eebd4148a33e